linode/nodepool/framework_models.go

```diff
@@ -0,0 +1,161 @@
+package nodepool
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/linode/linodego"
+	"github.com/linode/terraform-provider-linode/v2/linode/helper"
+)
+
+type NodePoolModel struct {
```

Comment:
@zliang-akamai Would you mind taking a look at the flatten functions here for preserveKnown, etc. since you've done a lot of work on that recently?

Reply:
Yep, I will take a look

Reply:
@zliang-akamai I need your advice regarding Autoscaler []NodePoolAutoscalerModel `tfsdk:"autoscaler"`
I have not found any similar example which uses preserveKnown

Reply:
Hi @michael-sh-guardicore, since `autoscaler` is not a computed block, it will never be a unknown value, and we can safely always ignore it where we want to preserve known values (e.g. `Create` and `Update` function). I put it together in a PR, and feel free to let me know if you would need to make any change on it.
https://github.com/michael-sh-guardicore/terraform-provider-linode/pull/1/